src/ItemMeasurer.js

```diff
@@ -46,44 +45,163 @@ type ItemMeasurerProps = {|
+const scrollableWrapperStyle = {
+  position: 'absolute',
+  flex: '0 0 auto',
+  overflow: 'hidden',
+  visibility: 'hidden',
+  zIndex: '-1',
+  width: '100%',
+  height: '100%',
+  left: '0px',
+  top: '0px',
+};
+
+const expandShrinkContainerStyles = {
+  flex: '0 0 auto',
+  overflow: 'hidden',
+  zIndex: '-1',
+  visibility: 'hidden',
+  left: '-9px',
+  bottom: '-8px',
+  right: '-8px',
+  top: '-9px',
+};
+
+const expandShrinkStyles = {
+  position: 'absolute',
+  flex: '0 0 auto',
+  visibility: 'hidden',
+  overflow: 'scroll',
+  zIndex: '-1',
+  width: '100%',
+  height: '100%',
+};
+
+const shrinkChildStyle = {
+  position: 'absolute',
+  height: '200%',
+  width: '200%',
+};
+
 export default class ItemMeasurer extends Component<ItemMeasurerProps, void> {
   _node: HTMLElement | null = null;
   _resizeObserver: ResizeObserver | null = null;
+  _resizeSensorExpand = React.createRef();
+  _resizeSensorShrink = React.createRef();
 
   componentDidMount() {
     const node = ((findDOMNode(this): any): HTMLElement);
     this._node = node;
-
     // Force sync measure for the initial mount.
     // This is necessary to support the DynamicSizeList layout logic.
     this._measureItem(true);
-
-    this._resizeObserver = new MutationObserver(this._onResize);
-    this._resizeObserver.observe(node, {
-      childList: true,
-      characterData: true,
-      subtree: true,
-    });
+    if (this.props.size) {
+      // Don't wait for positioning scrollbars when we have size
+      // This is needed triggering an event for remounting a post
+      this.positionScrollBars();
+    }
   }
 
   componentDidUpdate(prevProps) {
     if (prevProps.width !== this.props.width) {
       this._onResize();
     }
+
+    if (
+      (prevProps.size === 0 && this.props.size !== 0) ||
+      prevProps.size !== this.props.size
+    ) {
+      this.positionScrollBars();
+    }
   }
 
+  positionScrollBars = (height = this.props.size, width = this.props.width) => {
+    //we are position these hiiden div scroll bars to the end so they can emit
+    //scroll event when height in the div changes
+    //Heavily inspired from https://github.com/marcj/css-element-queries/blob/master/src/ResizeSensor.js
+    //and https://github.com/wnr/element-resize-detector/blob/master/src/detection-strategy/scroll.js
+    //For more info http://www.backalleycoder.com/2013/03/18/cross-browser-event-based-element-resize-detection/#comment-244
+
+    if (typeof this._resizeSensorExpand.current.scrollBy === 'function') {
+      this._resizeSensorExpand.current.scrollBy(height + 27, width + 27);
+
+      this._resizeSensorShrink.current.scrollBy(
+        2 * height + 17,
+        2 * width + 17
+      );
+    } else {
+      this._resizeSensorExpand.current.scrollLeft = width + 27;
+      this._resizeSensorExpand.current.scrollTop = height + 27;
+      this._resizeSensorShrink.current.scrollTop = 2 * height + 17;
+      this._resizeSensorShrink.current.scrollLeft = 2 * width + 17;
```

Comment:
nit: Can we make 27 and 17 as constant and add comment so that we know how/when to change it in the future?

Reply:
Nice catch. i thought to comment that but i forgot about it. I will add the comment for now. I am planning on making it dynamic and set the values on init later

Reply:
Done.